_posts/2020-11-02-dynamically-changeable-logging.md

```diff
@@ -0,0 +1,163 @@
+
+The entry `monitorInterval` must stay in place.
+When it is removed, the configuration will not be reloaded dynamically anymore.
+To apply changes in such a case, the pod needs to be rolled. 
+
+The second way to change `ConfigMap` is to modify the file `install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml`, and apply the changes by this command:
+
+```
+$ kubectl apply -f install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml
+```
+
+The same restrictions apply.
+The entry `monitorInterval` must stay in place.
+
+The change in logging takes some time to be applied. 
+You can find out more in the implementation details section of this post.
+
+### Topic Operator, User Operator and Kafka resources
+
+The approach to changing the logging configuration of User Operator, Topic Operator, Kafka Bridge, Kafka brokers, Kafka Connect and Kafka MirrorMaker2 did not change.
+The logging is still configurable via `inline` or `external` logging section of custom resource specification.
+
+Example of `inline` logging.
+```yaml
+...
+spec:
+  logging:
+    type: inline
+      loggers:
+        kafka.root.logger.level = INFO
+...
+```
+
+Example of external logging.
+```yaml
+...
+spec:
+  logging:
+    type: external
+      name: my-external-config-map
+...
+```
+
+Only loggers can be changed dynamically.
+Logging appenders are changed using an external `ConfigMap` and the rolling update is triggered.
+
+This does not apply for Kafka Bridge, User operator, Topic operator and Cluster operator.
+Because these use `log4j2` framework, even logging appenders are changeable dynmaicaly.
+
+
+## Verifying logging levels are set correctly
+
+In most cases, it takes a few seconds to change the logging level, and then you can observe the change in the logs.
+If you want to check the loggers and logging levels are changed, you can use these commands.
+
+For Kafka brokers:
+
+```
+$ kubectl  exec <kafka-cluster-name>-kafka-0 -- bin/kafka-configs.sh --bootstrap-server <kafka-cluster-name>-kafka-bootstrap:9092 --entity-type broker-loggers --entity-name 0 --describe
+```
+
+For Kafka Connect you can get loggers information by accessing Connect API service:
+
+```
+$ kubectl exec -ti <kafka-connect-pod-name> -- curl http://localhost:8083/admin/loggers
+```
+
+Since Kafka MirrorMaker2 is based on the Kafka Connect, getting its loggers is very similar:
+```
+$ kubectl exec -ti <kafka-mirror-maker-2-pod-name> -- curl http://localhost:8083/admin/loggers
+```
+
+Strimzi Cluster Operator, Topic Operator, Entity Operator and Strimzi Kafka Bridge use automatic reloading of logging configuration.
+The only way to check the logging configuration of these components is by looking into the pod for the file on the path `/opt/strimzi/custom-config/log4j2.properties`.
+It should contain the configuration and the `monitorInterval` property.
+Note that reconciliation time and `monitorInterval` value affect the time logging configuration to be reloaded.
```

Comment:
Should we add a line explaining that it also depends on the time that Kubernetes needs to refresh the configmap as volume in the pod?
See https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically